pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdConsume.java

```diff
@@ -404,4 +427,35 @@ public void close() {
         private static final Logger log = LoggerFactory.getLogger(ConsumerSocket.class);
 
     }
+
+    private static class DataConsistencyChecker {
+        private final ConcurrentHashMap<String, AtomicLong> messagesPerKey = new ConcurrentHashMap<>();
```

Comment:
There is ConcurrentBitSet in Pulsar

Reply:
I would like to count the duplicates,  ConcurrentBitSet would help only in order to check for duplicates, but if I want to dump the number of messages per key it won't work

Reply:
I'm sorry, because I saw you throw an exception directly above, and only record a duplicate record of one message.

Reply:
We can only record duplicate MessageId

Reply:
If you do not set the failOnDeuplicateKey option you can simply run the client and dump a summary in the end.
This way you can make more analysis.
So it is good to also have the count per key.

I am interested in keys and not in messageid.

Probably having a per message id count will be useful as well, but that is another usecase